packages/compiler/src/render3/view/compiler.ts

```diff
@@ -423,12 +423,20 @@ export function createBaseDirectiveTypeParams(meta: R3DirectiveMetadata): o.Type
     typeWithParameters(meta.type.type, meta.typeArgumentCount),
     selectorForType !== null ? stringAsType(selectorForType) : o.NONE_TYPE,
     meta.exportAs !== null ? stringArrayAsType(meta.exportAs) : o.NONE_TYPE,
-    o.expressionType(stringMapAsLiteralExpression(meta.inputs)),
+    o.expressionType(getInputsTypeExpression(meta)),
     o.expressionType(stringMapAsLiteralExpression(meta.outputs)),
     stringArrayAsType(meta.queries.map(q => q.propertyName)),
   ];
 }
 
+function getInputsTypeExpression(meta: R3DirectiveMetadata): o.Expression {
+  // TODO(required-inputs): expand this to generate the new object literal syntax.
+  return o.literalMap(Object.keys(meta.inputs).map(key => {
+    const value = meta.inputs[key];
+    return ({key, value: o.literal(typeof value === 'string' ? value : value.alias), quoted: true});
```

Comment:
This appears to have been pulled from an arrow function's expression body, but now that it's a return statement the parens are redundant:

```suggestion
    return {key, value: o.literal(typeof value === 'string' ? value : value.alias), quoted: true};
```